File: convert-dark-to-light.js
#!/usr/bin/env node
/**
 * Script pour convertir tous les écrans avec fond noir en fond blanc
 */
const fs = require('fs');
const path = require('path');

const BASE_DIR = path.join(__dirname, 'stitch_onboarding_ia_mesures 3');

// Fonction pour traiter un fichier
function processFile(filePath) {
    try {
        let content = fs.readFileSync(filePath, 'utf8');
        const originalContent = content;
        
        // Remplacements des classes dark:
        const replacements = [
            // Backgrounds
            [/dark:bg-background-dark/g, 'bg-white'],
            [/dark:bg-\[#1e293b\]/g, 'bg-white'],
            [/dark:bg-\[#131022\]/g, 'bg-white'],
            [/dark:bg-\[#0f172a\]/g, 'bg-white'],
            [/dark:bg-gray-900/g, 'bg-white'],
            [/dark:bg-gray-800/g, 'bg-gray-50'],
            [/dark:bg-gray-700/g, 'bg-gray-100'],
            [/dark:bg-slate-800/g, 'bg-gray-50'],
            [/dark:bg-slate-900/g, 'bg-white'],
            
            // Text colors
            [/dark:text-white/g, 'text-gray-900'],
            [/dark:text-slate-400/g, 'text-gray-600'],
            [/dark:text-gray-400/g, 'text-gray-600'],
            [/dark:text-gray-300/g, 'text-gray-700'],
            [/dark:text-gray-500/g, 'text-gray-700'],
            
            // Borders
            [/dark:border-gray-700/g, 'border-gray-300'],
            [/dark:border-gray-800/g, 'border-gray-200'],
            
            // Hover states
            [/dark:hover:bg-gray-800/g, 'hover:bg-gray-100'],
            [/dark:hover:bg-gray-700/g, 'hover:bg-gray-200'],
            [/dark:hover:text-gray-300/g, 'hover:text-gray-700'],
            [/dark:hover:text-primary/g, 'hover:text-primary'],
            
            // Rings
            [/dark:ring-white\/5/g, 'ring-gray-200/10'],
            [/dark:ring-white\/10/g, 'ring-gray-200/20'],
            
            // Shadows
            [/dark:shadow-\[0_8px_30px_rgb\(0,0,0,0\.3\)\]/g, 'shadow-lg'],
            
            // Opacity backgrounds
            [/dark:bg-primary\/10/g, 'bg-primary/5'],
            [/dark:bg-primary\/20/g, 'bg-primary/10'],
            
            // Focus ring offset
            [/dark:focus:ring-offset-gray-900/g, 'focus:ring-offset-white'],
        ];
        
        // Appliquer tous les remplacements
        replacements.forEach(([pattern, replacement]) => {
            content = content.replace(pattern, replacement);
        });
        
        // Supprimer les classes dark: restantes (en gardant l'espace avant)
        content = content.replace(/\s+dark:[^\s"']+/g, '');
        
        // Remplacer class="dark" par class="light" dans les balises html
        content = content.replace(/<html\s+class="dark"/g, '<html class="light"');
        content = content.replace(/<html\s+class='dark'/g, "<html class='light'");
        
        // Nettoyer les espaces multiples
        content = content.replace(/\s{2,}/g, ' ');
        content = content.replace(/\s+class="/g, ' class="');
        content = content.replace(/\s+class='/g, " class='");
        
        // Écrire seulement si le contenu a changé
        if (content !== originalContent) {
            fs.writeFileSync(filePath, content, 'utf8');
            return true;
        }
        return false;
    } catch (error) {
        console.error(`Erreur lors du traitement de ${filePath}:`, error.message);
        return false;
    }
}

// Fonction récursive pour trouver tous les fichiers code.html
function findHtmlFiles(dir) {
    const files = [];
    try {
        const entries = fs.readdirSync(dir, { withFileTypes: true });
        for (const entry of entries) {
            const fullPath = path.join(dir, entry.name);
            if (entry.isDirectory()) {
                files.push(...findHtmlFiles(fullPath));
            } else if (entry.isFile() && entry.name === 'code.html') {
                files.push(fullPath);
            }
        }
    } catch (error) {
        console.error(`Erreur lors de la lecture de ${dir}:`, error.message);
    }
    return files;
}

// Fonction principale
function main() {
    if (!fs.existsSync(BASE_DIR)) {
        console.error(`Erreur: Le dossier ${BASE_DIR} n'existe pas`);
        process.exit(1);
    }
    
    const htmlFiles = findHtmlFiles(BASE_DIR);
    console.log(`Trouvé ${htmlFiles.length} fichiers HTML à traiter...\n`);
    
    let modified = 0;
    for (const filePath of htmlFiles) {
        if (processFile(filePath)) {
            modified++;
            const relativePath = path.relative(BASE_DIR, filePath);
            console.log(`✓ Modifié: ${relativePath}`);
        }
    }
    
    console.log(`\n✅ Traitement terminé: ${modified} fichiers modifiés sur ${htmlFiles.length}`);
}

main();

File: convert-dark-to-light.py
#!/usr/bin/env python3
"""
Script pour convertir tous les écrans avec fond noir en fond blanc
"""
import os
import re
from pathlib import Path

# Dossier contenant les maquettes
BASE_DIR = Path("stitch_onboarding_ia_mesures 3")

# Remplacements à effectuer
replacements = [
    # Classes dark:bg-background-dark → bg-white ou bg-background-light
    (r'dark:bg-background-dark', 'bg-white'),
    (r'dark:bg-\[#1e293b\]', 'bg-white'),
    (r'dark:bg-\[#131022\]', 'bg-white'),
    (r'dark:bg-\[#0f172a\]', 'bg-white'),
    
    # Classes dark:bg-gray-* → bg-white ou bg-gray-50
    (r'dark:bg-gray-900', 'bg-white'),
    (r'dark:bg-gray-800', 'bg-gray-50'),
    (r'dark:bg-gray-700', 'bg-gray-100'),
    (r'dark:bg-slate-800', 'bg-gray-50'),
    (r'dark:bg-slate-900', 'bg-white'),
    
    # Classes dark:text-white → text-gray-900
    (r'dark:text-white', 'text-gray-900'),
    (r'dark:text-slate-400', 'text-gray-600'),
    (r'dark:text-gray-400', 'text-gray-600'),
    (r'dark:text-gray-300', 'text-gray-700'),
    (r'dark:text-gray-500', 'text-gray-700'),
    
    # Classes dark:border-* → border-gray-300
    (r'dark:border-gray-700', 'border-gray-300'),
    (r'dark:border-gray-800', 'border-gray-200'),
    
    # Classes dark:hover:*
    (r'dark:hover:bg-gray-800', 'hover:bg-gray-100'),
    (r'dark:hover:bg-gray-700', 'hover:bg-gray-200'),
    (r'dark:hover:text-primary', 'hover:text-primary'),
    (r'dark:hover:text-gray-300', 'hover:text-gray-700'),
    
    # Classes dark:ring-*
    (r'dark:ring-white/5', 'ring-gray-200/10'),
    (r'dark:ring-white/10', 'ring-gray-200/20'),
    
    # Classes dark:shadow-*
    (r'dark:shadow-\[0_8px_30px_rgb\(0,0,0,0\.3\)\]', 'shadow-lg'),
    (r'dark:shadow-\[.*?\]', 'shadow-lg'),
    
    # Classes dark:bg-* avec opacité
    (r'dark:bg-primary/10', 'bg-primary/5'),
    (r'dark:bg-primary/20', 'bg-primary/10'),
    
    # Supprimer les classes dark: restantes
    (r'\s+dark:[^\s]+', ''),  # Supprime les classes dark: restantes
]

def process_file(file_path):
    """Traite un fichier HTML"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Appliquer tous les remplacements
        for pattern, replacement in replacements:
            content = re.sub(pattern, replacement, content)
        
        # Remplacer class="dark" par class="light" dans les balises html
        content = re.sub(r'<html\s+class="dark"', '<html class="light"', content)
        content = re.sub(r'<html\s+class=\'dark\'', '<html class="light"', content)
        
        # Nettoyer les espaces multiples créés par les suppressions
        content = re.sub(r'\s{2,}', ' ', content)
        content = re.sub(r'\s+class="', ' class="', content)
        content = re.sub(r'\s+class=\'', ' class=\'', content)
        
        # Écrire seulement si le contenu a changé
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        return False
    except Exception as e:
        print(f"Erreur lors du traitement de {file_path}: {e}")
        return False

def main():
    """Fonction principale"""
    if not BASE_DIR.exists():
        print(f"Erreur: Le dossier {BASE_DIR} n'existe pas")
        return
    
    html_files = list(BASE_DIR.rglob("code.html"))
    print(f"Trouvé {len(html_files)} fichiers HTML à traiter...")
    
    modified = 0
    for file_path in html_files:
        if process_file(file_path):
            modified += 1
            print(f"✓ Modifié: {file_path.relative_to(BASE_DIR.parent)}")
    
    print(f"\n✅ Traitement terminé: {modified} fichiers modifiés sur {len(html_files)}")

if __name__ == "__main__":
    main()
